portal-api/src/scheduler/jobs/send-report-due-emails-job/index.js

```diff
@@ -0,0 +1,113 @@
+const { format, isSameDay } = require('date-fns');
+const { getReportDueDate, getReportPeriodMonthAndYear, getFormattedReportPeriod } = require('../helpers/utilisation-report-helpers');
+const api = require('../../../v1/api');
+const { hasValue, isValidEmail } = require('../../../utils/string');
+const sendEmail = require('../../../external-api/send-email');
+const EMAIL_TEMPLATE_IDS = require('../../../constants/email-template-ids');
+
+const { PORTAL_REPORT_DUE_EMAIL_SCHEDULE } = process.env;
+
+const EMAIL_NAME = 'GEF utilisation report due';
+
+/**
+ * Sends the email to the specified bank when a valid payment officer team email is present
+ * @param bank {object} - the bank to send the email to
+ * @param reportPeriod {string} - the formatted reporting period that will appear in the email
+ */
+const sendEmailForBank = async (bank, reportPeriod) => {
+  const { name: bankName, paymentOfficerTeam } = bank;
+  const paymentOfficerTeamEmail = paymentOfficerTeam?.email;
+
+  try {
+    if (!hasValue(paymentOfficerTeamEmail)) {
+      console.warn(`Not sending ${EMAIL_NAME} email to '${bankName}' - no payment officer team email set`);
+    } else if (!isValidEmail(paymentOfficerTeamEmail)) {
+      console.error(`Failed to send ${EMAIL_NAME} email to '${bankName}' - invalid payment officer team email '${paymentOfficerTeamEmail}'`);
+    } else {
+      await sendEmail(EMAIL_TEMPLATE_IDS.UTILISATION_REPORT_DUE_TODAY, paymentOfficerTeamEmail, {
+        recipient: paymentOfficerTeam.teamName,
+        reportPeriod,
+      });
+      console.info(`Successfully sent ${EMAIL_NAME} email to '${bankName}'`);
+    }
+  } catch (error) {
+    console.error(`Failed to send ${EMAIL_NAME} for bank '${bankName}':`, error);
+  }
+};
+
+/**
+ * Checks if a utilisation report has already been submitted by the specified bank for the current reporting period
+ * @param bank {object} - the bank to check
+ * @returns {Promise<boolean>}
+ */
+const getIsReportSubmitted = async (bank) => {
+  // TODO FN-1163 - check how this endpoint has been updated
+  const reportsResponse = await api.getUtilisationReports(bank.id);
+
+  if (reportsResponse?.status !== 200) {
+    const error = reportsResponse?.data ?? 'unknown error';
+    throw new Error(`Failed to get utilisation reports for ${bank.name} (bank ID: ${bank.id}): %s`, error);
+  }
+
+  const reportPeriod = getReportPeriodMonthAndYear();
+  return reportsResponse.data.some((report) => report.month === reportPeriod.month && report.year === reportPeriod.year);
```

Comment:
Could we maybe destructure `reportsResponse` from line 45 into `{ status, data }` and change the condition to `status && status !== 200`?